<a href="https://colab.research.google.com/github/WeiyangBellaBao/my-first-project/blob/main/Rideshare_Reviews_Analyst_Output.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# About

Here is my work for the review of the client's idea.

I used the original dataset supplied by the client at the URL below:

```
https://raw.githubusercontent.com/Btibert3/datasets/master/rideshare.csv
```

In [ ]:
# imports
import pandas as pd
from sklearn.feature_extraction.text import CountVectorizer
from sklearn.tree import DecisionTreeClassifier
from sklearn import metrics

In [ ]:
# bring in the data
df = pd.read_csv("https://raw.githubusercontent.com/Btibert3/datasets/master/rideshare.csv")

In [ ]:
# shape
df.shape

(1000, 2)

In [ ]:
# sample
df.sample(3)

ride_review  ride_rating
651  I Boston business Today I send Uber pick lunch...            1
318  I uber driver I money get customers Rating loa...            1
301  Uber customer service non existent They respon...            1

In [ ]:
# data cleaning
df.dropna(inplace=True)

In [ ]:
# tokenize the text
cv = CountVectorizer(max_features=30)
dtm = cv.fit_transform(df.ride_review).toarray()

In [ ]:
# make a dataset for the model
df2 = pd.DataFrame(dtm, columns=cv.get_feature_names_out())


In [ ]:
# quick confirmation
df2.sample(3)

account  app  back  car  card  charged  ...  they  time  told  trip  uber  use
463        0    0     0    0     0        0  ...     2     2     0     0     5    1
580        0    0     1    0     0        0  ...     0     0     2     0     3    0
689        0    0     0    1     0        0  ...     0     0     0     3     3    0

[3 rows x 30 columns]

In [ ]:
# fit the model
tree = DecisionTreeClassifier(min_samples_leaf=1000, min_samples_split=2000, random_state=820)
tree.fit(df2, df.ride_rating)

DecisionTreeClassifier(min_samples_leaf=1000, min_samples_split=2000,
                       random_state=820)

In [ ]:
# how well do the model do?
cr = metrics.classification_report(df.ride_rating, tree.predict(df2))
print(cr)

              precision    recall  f1-score   support

           1       0.77      1.00      0.87       766
           2       0.00      0.00      0.00        54
           3       0.00      0.00      0.00        42
           4       0.00      0.00      0.00        47
           5       0.00      0.00      0.00        91

    accuracy                           0.77      1000
   macro avg       0.15      0.20      0.17      1000
weighted avg       0.59      0.77      0.66      1000



/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1308: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1308: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1308: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


 - 77% accurate
 - perfect recall on the lowest rating of 1!


 > My opinion is that we can work with the client to deploy the sampling strategy via the app, as we can predict the ratings well enough when there isn't one supplied.